### DISCUSSION TITLE
Rearranging glowroot project.

### DISCUSSION THREAD

**Discussion by Neyos1997** [glowroot/glowroot#1012] [General]
Hello everyone , i have some issues since i am new to this api and controlling it's behavior , 
first of all i  built the cloned glowroot project  several jars were generated .  
i would like to deploy glowroot within my application   with some changes for the ui . 
untill this moment i know that the jar under /dist is the one to integrate  with the project , else when it comes to do some changes to the ui i don't know which file to handle , and how to integrate the new ui if i am able to to some changes in it , my goal is to make the errors section invisible .  
if i only deploy the jar inder /dist how can i bind the changed ui in it ? 
How to change the navbar  to make errors section invisible ? 
how would the jar under /dist would manage to take in consideraton the changes in th ui ?!  i used some decompiler to see the code no ui indication there  . 
Thanks for your help

**Comment by nowheresly** [comment]
Hi @Neyos1997 
if you want to build a glowroot central wothout the error tab, I suggest that you modify the file index.html in the ui/app folder

https://github.com/glowroot/glowroot/blob/8f545ddc159e362a577300d8a7aec8dc0340769d/ui/app/index.html#L113

remove the block:
```
          <div gt-navbar-item
               gt-display="Errors"
               gt-item-name="error"
               gt-url="error/messages{{queryString(true, true)}}"
               gt-click="collapseIfNeeded()"
               ng-if="layout.showNavbarError">
          </div>
```

**Reply by Neyos1997** [reply]
Hi @Nowheresly  Thanks for your answer . 

as far as the documentation shows , the glowroot should be used is the one under /dist , i would like to deploy glowroot with my application in a simple way . like i would like to package the whole glowroot with its ui in a jar then deploy it within the application . 
it is just a little bit complicated to me after cloning  and building pratically every file has /target and a jar i am literally with no clue

**Comment by Never-lab** [comment]
Quick clarification on deployment paths (since the thread mixes UI hacking and packaging):

- **Typical production use:** attach `glowroot.jar` as a **`-javaagent`** to your application JVM (embedded mode, local H2 + UI), or point the agent at a **Central collector** (Cassandra). You do **not** embed Glowroot as a Maven dependency inside your app.
- **The `/dist` zip** is the supported distribution artifact after `mvn clean install`.
- **UI changes:** build from the `ui/` module; for backend+UI together use embedded or Central as documented in CONTRIBUTING — the UI is not a separate runtime JAR you bolt onto a random app JAR.

Happy to expand in **Q&A** if you share embedded vs central and your target deployment (Tomcat WAR, Spring Boot fat JAR, etc.).
